cloudfunctions2_basic/main.tf

```diff
@@ -1,10 +1,10 @@
 # [START functions_v2_basic]
-locals {
-  project = "my-project-name" # Google Cloud Platform Project ID
+resource "random_id" "prefix" {
+  byte_length = 8
 }
 
 resource "google_storage_bucket" "bucket" {
-  name     = "${local.project}-gcf-source"  # Every bucket name must be globally unique
+  name     = "${random_id.prefix.hex}-gcf-source"  # Every bucket name must be globally unique
   location = "US"
   uniform_bucket_level_access = true
 }
```

Comment:
Re: `function-source.zip` - instead of adding the functions inside of a zip file, could we instead add the files in a sub-directory and then use the `archive_file` terraform resource? 

For example, move `index.js` and `package.json` to `functions/hello-world/` and then:
```
data "archive_file" "archive" {
  type        = "zip"
  output_path = "/tmp/function-source.zip"
  source_dir  = "functions/hello-world"
}
resource "google_storage_bucket_object" "object" {
  name   = "function-source.zip"
  bucket = google_storage_bucket.bucket.name
  source = data.archive_file.archive.output_path  # Add path to the zipped function source code
}
```


It adds a little more complexity to the sample but could be useful for showing one method of including those functions in source control along side their terraformed cloud function resources.

Reply:
Opened an issue instead for follow up if other sample maintainers agree.